Detections/MultipleDataSources/SecurityServiceRegistryACLModification.yaml

```diff
@@ -0,0 +1,86 @@
+
+id: 473d57e6-f787-435c-a16b-b38b51fa9a4b 
+name: Security Service Registry ACL Modification
+description: |
+    'Identifies attempts to modify registry ACL to evade security solutions. In the Solorigate attack, the attackers were found modifying registry permissions so services.exe cannot access the relevant registry keys to start the service.
+     The detection leverages Security Event as well as MDE data to identify when specific security services registry permissions are modified. 
+     Only some portions of this detection are related to Solorigate, it also includes coverage for some common tools that perform this activity. 
+     Reference on guidance for enabling registry auditing:
+     - https://docs.microsoft.com/windows/security/threat-protection/auditing/advanced-security-auditing-faq
+     - https://docs.microsoft.com/windows/security/threat-protection/auditing/appendix-a-security-monitoring-recommendations-for-many-audit-events
+     - https://docs.microsoft.com/windows/security/threat-protection/auditing/audit-registry
+     - https://docs.microsoft.com/windows/security/threat-protection/auditing/event-4670
+     - https://github.com/OTRF/Set-AuditRule 
+     - https://docs.microsoft.com/en-us/dotnet/api/system.security.accesscontrol.registryrights?view=dotnet-plat-ext-5.0
+     For the above event to be created the audit policy for the registry must have auditing enabled for Write DAC and/or Write Owner'
+severity: High
+requiredDataConnectors:
+  - connectorId: SecurityEvents
+    dataTypes:
+      - SecurityEvent 
+  - connectorId: MicrosoftThreatProtection
+    dataTypes:
+      - DeviceProcessEvents
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - DefenseEvasion
+relevantTechniques:
+  - T1562
+tags: 
+  - Solorigate
+query: |
+ 
+  let servicelist = dynamic(['Services\\HealthService', 'Services\\Sense', 'Services\\WinDefend', 'Services\\MsSecFlt', 'Services\\DiagTrack', 'Services\\SgrmBroker', 'Services\\SgrmAgent', 'Services\\AATPSensorUpdater' , 'Services\\AATPSensor', 'Services\\mpssvc']);
+  let filename = dynamic(["subinacl.exe",'SetACL.exe']);
+  let parameters = dynamic (['/deny=SYSTEM', '/deny=S-1-5-18', '/grant=SYSTEM=r', '/grant=S-1-5-18=r', 'n:SYSTEM;p:READ', 'n1:SYSTEM;ta:remtrst;w:dacl']);
```

Comment:
This is good for now, but could be fragile.  After this version is published verify that a regex match is not needed to handle if there are spaces include in the parameters being passed in.  Such as "/deny = SYSTEM" or "n:SYSTEM; p:READ"

Reply:
Ack. Will work on it